src/pages/bulkImport/BulkReportForm.jsx

```diff
@@ -91,6 +140,47 @@ export default function BulkReportForm({ assetReferences }) {
   const sightingFieldSchemas = useSightingFieldSchemas();
   const encounterFieldSchemas = useEncounterFieldSchemas();
 
+  const multiselectFieldIds = reduce(
+    [...sightingFieldSchemas, ...encounterFieldSchemas],
+    (memo, currentSchema) => {
+      if (currentSchema?.fieldType === 'multiselect')
+        return [...memo, currentSchema?.id];
+      return memo;
+    },
+    [],
+  );
+
+  const matchingCustomMultiselects = filter(
+    availableFields,
+    field => {
+      const matches = field?.key?.match(
+        /custom-(encounter|sighting|individual)-(.*)/,
+      ); // I imagine that there's a way to generalize this to work with deriveCustomFieldPrefix, but I played around with new regExp() using variables, and couldn't seem to get it to work. Would love some help with this.
+      return multiselectFieldIds.includes(matches && matches[2]);
+    },
+  );
+
+  const multiselectCustomFieldHooks = reduce(
```

Comment:
check out `_.keyBy` for a good opportunity to really simplify this code

Reply:
I ended up having to use both keyBy and mapValues to solve this problem.

Reply:
Change in https://github.com/WildMeOrg/codex-frontend/pull/431/commits/aeaa7cf1199b0121a3fcb6c48a39600a45ce0faf